server/base/src/main/java/org/apache/accumulo/server/util/ServerBulkImportStatus.java

```diff
@@ -37,12 +39,22 @@
 
   public void updateBulkImportStatus(List<String> files, BulkImportState state) {
     for (String file : files) {
-      BulkImportStatus initial = new BulkImportStatus(System.currentTimeMillis(), file, state);
-      status.putIfAbsent(file, initial);
-      initial = status.get(file);
-      if (initial != null) {
-        initial.state = state;
-      }
+      updateFile(file, state);
+    }
+  }
+
+  private void updateFile(String file, BulkImportState state) {
+    BulkImportStatus initial = new BulkImportStatus(System.currentTimeMillis(), file, state);
+    status.putIfAbsent(file, initial);
+    initial = status.get(file);
+    if (initial != null) {
+      initial.state = state;
+    }
+  }
```

Comment:
This could be made to update the concurrent map atomically:

```suggestion
  private void updateFile(String file, BulkImportState state) {
    status.compute(file, (key, currentStatus) -> {
      if (currentStatus == null) {
        return new BulkImportStatus(System.currentTimeMillis(), file, state);
      }
      currentStatus.state = state;
      return currentStatus;
    });
  }
```

Reply:
I figured there was a way but couldn't find the `compute` method. Nice!